### PR TITLE
Add serializer to handshake response and save handshake data on the session.

### PR DESCRIPTION
**Pull request by leohahn**

### CODE REVIEW

File: service/handler.go [hunk 1]
```diff
@@ -207,14 +209,27 @@ func (h *HandlerService) Handle(conn net.Conn) {
 }
 
 func (h *HandlerService) processPacket(a *agent.Agent, p *packet.Packet) error {
+	fmt.Println("PROCESSING PAKCET MAMN")
```
Comment (Henrod) [inline service/handler.go]: Should this be here?

File: service/handler.go [hunk 2]
```diff
@@ -207,14 +209,27 @@ func (h *HandlerService) Handle(conn net.Conn) {
...
-		a.SetStatus(constants.StatusHandshake)
 		logger.Log.Debugf("Session handshake Id=%d, Remote=%s", a.Session.ID(), a.RemoteAddr())
 
+		// Parse the json sent with the handshake by the client
+		handshakeData := &session.HandshakeData{}
+		err := json.Unmarshal(p.Data, handshakeData)
+
+		if err != nil {
+			a.SetStatus(constants.StatusClosed)
+			println("INVALID HANDSHAKE DATA BOI")
```
Comment (Henrod) [inline service/handler.go]: Should this be here?

File: session/session.go [hunk 3]
```diff
@@ -58,6 +58,22 @@ var (
 	SessionCount int64
 )
 
+// HandshakeClientData represents information about the client sent on the handshake.
+type HandshakeClientData struct {
+	Platform    string `json:"platform"`
+	LibVersion  string `json:"lib_version"`
```
Comment (Henrod) [inline session/session.go]: I think the json format should be libVersion (camel case). But if the client also is lib_version, that's not a big problem.

File: session/session.go [hunk 4]
```diff
@@ -69,6 +85,7 @@ type Session struct {
 	lastTime          int64                  // last heartbeat time
 	entity            NetworkEntity          // low-level network entity
 	data              map[string]interface{} // session data store
+	handshakeData     *HandshakeData         // handshake data received by the client
```
Comment (Henrod) [inline session/session.go]: Why didn't you put the handshake data on session's data?

File: service/handler.go (hunk 1 above)
Comment (leohahn) [reply service/handler.go]: Oops, will remove that.

File: session/session.go (hunk 3 above)
Comment (leohahn) [reply session/session.go]: It can be camel case as well. I can change it.

File: service/handler.go (hunk 2 above)
Comment (cscatolini) [reply service/handler.go]: best log 🤣

File: session/session.go (hunk 4 above)
Comment (leohahn) [reply session/session.go]: That way the user would have to keep in mind of the handshake data when changing the session data. For example, clearing the session data would also clear the handshake data, this not being very obvious.

**Pull request merged**